src/NuGet.Services.Metadata.Catalog.Monitoring/Utility/SafeExceptionConverter.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Newtonsoft.Json;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring
+{
+    /// <summary>
+    /// <see cref="JsonConverter"/> for converting exceptions safely.
+    /// If the exception fails to deserialize, returns an <see cref="ExceptionDeserializationException"/> instead of failing.
+    /// </summary>
+    public class SafeExceptionConverter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType)
+        {
+            return typeof(Exception).IsAssignableFrom(objectType);
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            try
+            {
+                return serializer.Deserialize(reader, objectType);
+            }
+            catch
+            {
+                // When deserializing the exception fails, we don't want to fail deserialization of the entire object.
+                // Return an ExceptionDeserializationException instead.
+                return new ExceptionDeserializationException();
```

Comment:
>ExceptionDeserializationException [](start = 27, length = 33)

Why would deserializing an exception fail? Is this a common case? If so, explain how. If not, should we maybe log something so that we can debug these issues?

Reply:
Basically, some exceptions don't implement the correct structure to be successfully deserialized by JSON.NET.

I encountered it in the past, and I found a StackOverflow post explaining the exact issue, but I can't seem to find it now.

Reply:
It's not something we can really debug, so I didn't add any logging here.

Reply:
Fair enough - please add a comment explaining this! :)